libraries/timeline-view-compose/src/main/java/com/trendyol/uicomponents/timelineviewcompose/Line.kt

```diff
@@ -0,0 +1,205 @@
+package com.trendyol.uicomponents.timelineviewcompose
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.LinearEasing
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.BlendMode
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.PathEffect
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import com.trendyol.uicomponents.timelineviewcompose.model.LineConfig
+import com.trendyol.uicomponents.timelineviewcompose.model.LineType
+import com.trendyol.uicomponents.timelineviewcompose.model.TimelineOrientation
+
+@Composable
+internal fun Line(
+    modifier: Modifier = Modifier,
+    config: LineConfig,
+    orientation: TimelineOrientation,
+    itemIndex: Int,
+) {
+    val pathEffect = when (config.lineType) {
+        is LineType.Dashed -> PathEffect.dashPathEffect(
+            config.lineType.intervals,
+            config.lineType.phase
+        )
+
+        else -> null
+    }
+
+    val itemSize = when (orientation) {
+        TimelineOrientation.HORIZONTAL -> DpSize(
+            width = config.size,
+            height = config.thickness
+        )
+
+        TimelineOrientation.VERTICAL -> DpSize(
+            width = config.thickness,
+            height = config.size
+        )
+    }
+
+    val animateLine = remember { Animatable(config.animation?.initialValue ?: 0f) }
+    if (config.animation != null) {
+        LaunchedEffect(animateLine) {
+            animateLine.animateTo(
+                targetValue = config.animation.targetValue,
+                animationSpec = tween(
+                    easing = LinearEasing,
```

Comment:
Should we take another easing `FastOutSlowInEasing`, `LinearOutSlowInEasing`, `FastOutLinearEasing`, `CubicBezierEasing`

Reply:
actually, I'm not sure, this is basic loading progress animation. but we can talk.